gordo_components/server/views/anomaly.py

```diff
@@ -152,86 +136,34 @@ def _process_base_response(self, response: Response, start_time: float = None):
         if start_time is None:
             start_time = timeit.default_timer()
 
-        # If something went wrong with the basic model view, no point going further
-        if not 200 <= response.status_code <= 299:
-            return response
-
-        # If client is accessing the anomaly endpoint where the model doesn't give
-        # back the transformed model input, we can't do anomaly formatting
-        if not self._output_matches_input_shape:
+        # To use this endpoint, we need a 'y' to calculate the errors.
+        # It has either come from client providing it in POST or from
+        # Influx during 'GET' part of `..common.extract_X_y` decorator
+        if g.y is None:
             message = {
-                "message": "Cannot perform anomaly detection: model output shape != input shape"
+                "message": "Cannot perform anomaly without 'y' to compare against."
             }
             return make_response((jsonify(message), 400))
 
+        # It is ok for y to be a subset of features in X, but we need at least one
+        # to compare against to calculate an error.
+        if not any(col in g.y.columns for col in [t.name for t in self.tags]):
+            message = {"message": "y is not a subset of X, cannot do anomaly detection"}
+            return make_response(jsonify(message), 400)
+
         # Now create an anomaly dataframe from the base response dataframe
-        anomaly_df = self.make_anomaly_df(self._data)
+        try:
+            anomaly_df = current_app.model.anomaly(g.X, g.y, frequency=self.frequency)
+        except AttributeError:
+            msg = {
+                "message": f"Model is not an AnomalyDetector, it is of type: {type(current_app.model)}"
+            }
+            return make_response(jsonify(msg), 400)
```

Comment:
Some other error code so we can distinguish it? `422 Unprocessable Entity` maybe?

Reply:
https://github.com/equinor/gordo-components/issues/400